File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    cnt = len(list(i for i in lottos if i in win_nums))
    cnt_zero = len(list(i for i in lottos if i == 0))
    cnt_max = cnt + cnt_zero    # cnt_min  = cnt
    answer = [6 if 7-cnt_max == 7 else 7-cnt_max, 6 if 7-cnt >= 6 else 7-cnt ]
    # print(answer)
    return answer

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] numbers = new int[46];
        
        for (int num : lottos) numbers[num]++; // 0~45
        
        int cnt = 0; // 일치하는 개수
        for (int num : win_nums) if(numbers[num] == 1) cnt++; // 1~45
        
        int[] rank = {6, 6, 5, 4, 3, 2, 1}; // 순위배열;
        return new int[]{rank[cnt + numbers[0]], rank[cnt]};
    }
}